ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -57,38 +64,105 @@
    modified times of the files unchanged. In linux use "cp -p" to maintain the file attributes.
    */
 
+  /**
+   * Config for the ConsistencyCheckerTool.
+   */
+  private static class ConsistencyCheckerToolConfig {
+
+    /**
+     * The path where the input is.
+     */
+    @Config("path.of.input")
+    final File pathOfInput;
+
+    /**
+     * The path to the hardware layout file. Needed if using
+     * {@link com.github.ambry.clustermap.StaticClusterAgentsFactory}.
+     */
+    @Config("hardware.layout.file.path")
+    @Default("")
+    final String hardwareLayoutFilePath;
+
+    /**
+     * The path to the partition layout file. Needed if using
+     * {@link com.github.ambry.clustermap.StaticClusterAgentsFactory}.
+     */
+    @Config("partition.layout.file.path")
+    @Default("")
+    final String partitionLayoutFilePath;
+
+    /**
+     * A comma separated list of blob IDs that the tool should operate on. Leaving this empty indicates that the tool
+     * should work on all blobs.
+     */
+    @Config("filter.set")
+    @Default("")
+    final Set<String> filterSet;
+
+    /**
+     * The number of index entries to process every second.
+     */
+    @Config("index.entries.to.process.per.sec")
+    @Default("Long.MAX_VALUE")
```

Comment:
can we set reasonable defaults and not Long.MAX. We shouldn't allow user's to un-intentionally execute this tool in full throttle.

Reply:
I actually think `Long.MAX_VALUE` is the reasonable default when this is triggered from the `main()` function since I expect the `main()` to be used in debugging scenarios and I would want the tool to run as fast as possible without being forced to change properties. When using the API version (i.e. maybe as a part of something that does this regularly), appropriate values can be passed.